src/legacy/core_plugins/tile_map/public/components/wms_options.tsx

```diff
@@ -23,23 +23,26 @@ import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
 
 import { toastNotifications } from 'ui/notify';
-import { VisOptionsSetValue } from 'ui/vis/editors/default';
 import { TmsLayer } from 'ui/vis/map/service_settings';
-import { ExtendedVisOptionsProps } from '../../../kbn_vislib_vis_types/public/utils/with_injected_dependencies';
 import { SelectOption } from '../../../kbn_vislib_vis_types/public/controls/select';
 import { SwitchOption } from '../../../kbn_vislib_vis_types/public/controls/switch';
 import { WmsInternalOptions } from './wms_internal_options';
+import { TileMapOptionsProps } from './tile_map_options';
+import { TileMapVisParams } from '../types';
 
 const mapLayerForOption = ({ id }: TmsLayer) => ({ text: id });
 
-function WmsOptions({ serviceSettings, stateParams, setValue, vis }: ExtendedVisOptionsProps) {
+function WmsOptions({ serviceSettings, stateParams, setValue, vis }: TileMapOptionsProps) {
```

Comment:
This can't use `TileMapOptionsProps` since the comp will be used in `ReionMapOptions` too

Reply:
Would you be fine if you pull this out in your region map PR into a base type, since I don't have the region map props yet available?